geocoder/hierarchy.cpp

```diff
@@ -199,6 +205,22 @@ void Hierarchy::IndexEntries()
   }
 }
 
+void Hierarchy::IndexStreet(Entry & e)
+{
+  CHECK_EQUAL(e.m_type, Type::Street, ());
+  size_t const t = static_cast<size_t>(e.m_type);
+  m_entriesByTokens[MakeIndexKey(e.m_address[t])].emplace_back(&e);
+
+  for (size_t i = 0; i < e.m_address[t].size(); ++i)
+  {
+    if (!search::IsStreetSynonym(strings::MakeUniString(e.m_address[t][i])))
+      continue;
+    auto addr = e.m_address[t];
```

Comment:
Possible to move this string outside the loop?

Reply:
I don't think so, see the next line.